src/main/java/com/leverx/learningmanagementsystem/config/security/SecurityAuditorAware.java

```diff
@@ -0,0 +1,33 @@
+package com.leverx.learningmanagementsystem.config.security;
+
+import org.springframework.data.domain.AuditorAware;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+
+import java.util.Optional;
+
+public class SecurityAuditorAware implements AuditorAware<String> {
+
+    @Override
+    public Optional<String> getCurrentAuditor() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if (!authentication.isAuthenticated()) {
+            return Optional.of("Anonymous");
+        }
```

Comment:
there is no need to check authentication, as this code will not be reached in case of authentication failure in security filter